gem/lib/pupperware/spec_helper.rb

```diff
@@ -29,8 +29,12 @@ def require_test_image()
   # General Ruby Helpers
   ######################################################################
 
+  # Returns hash with `status`, `stdout`, and `stderr` keys
+  # Use `result[:status].exitstatus` to get the exit code.
```

Comment:
I see in some tests we have suspicious usages of `result[:status]` that don't call `.exitstatus`, like this one: https://github.com/puppetlabs/puppet-runtime/blob/master/docker/puppet-runtime/spec/dockerfile_spec.rb#L12

Maybe things have changed over time since this method was introduced, but the `:status` value is not an integer, it's a `#<Process::Status: pid 6566 exit 0>` object.

I'm not sure what the `expect to eq(0)` call would result in on that object, but it doesn't seem like what we want either way.

Should we make some changes here around this?

- We can throw away the Process::Status object and make `:status` be the exit code value here, or
- We can continue to return this object and make users pull out the exitstatus field (what we're currently doing), or
- We can add a 4th field to this returned hash for the `pid`, which seems like the only reason to continue returning this Process::Status object

Reply:
My opinion would be to add the 4th `pid` field and not return intermediate objects from this method, so it's as easy as possible for callers and we don't leak implementation details out of this method (however minor they are).